cirq/contrib/routing/greedy.py

```diff
@@ -284,22 +301,19 @@ def apply_next_swaps(self):
                 ]
                 if len(candidate_swap_sets) == 1:
                     self.apply_swap(*candidate_swap_sets[0])
+                    return
 
-                    if list(
-                            self.remaining_dag.findall_nodes_until_blocked(
-                                self.acts_on_nonadjacent_qubits)):
-                        return
-                    else:
-                        break
-
-        frontier_edges = sorted(time_slices[0].edges)
-        self.bring_farthest_pair_together(frontier_edges)
+        self.apply_next_swaps(True)
 
     def route(self):
         self.apply_possible_ops()
+        empty_steps_remaining = self.max_num_empty_steps
         while self.remaining_dag:
-            self.apply_next_swaps()
-            self.apply_possible_ops()
+            self.apply_next_swaps(not empty_steps_remaining)
+            if self.apply_possible_ops():
```

Comment:
Would pull out self.apply_possible_ops() of the if into a variable remaining_ops or similar for better readability.

Reply:
Done